### DISCUSSION TITLE
How to imitate adding StyleInclude in App.axaml from code.

### DISCUSSION THREAD

**Discussion by MaxwellDAssistek** [AvaloniaUI/Avalonia#16700] [Q&A] [answered]
Hello,

I'm trying to use a library that provides its own styles (https://github.com/Deadpikle/AvaloniaProgressRing) however I am unable to change the main App.axaml file. Is there any way to add an application-wide StyleInclude from this library using C# code?

Thanks!

**Comment by maxkatz6** [comment, marked as answer]
Like in XAML, you need to create StyleInclude and set Source.
```xaml
App.Current.Styles.Add(new StyleInclude("avares://ThisProjectName")
{
    Source = new Uri("avares://AvaloniaProgressRing/Styles/ProgressRing.xaml")
})
```

**Comment by saeed-neorelm** [comment]
Hi, apparently that doesn't work with AOT enabled

**Reply by maxkatz6** [reply]
It's a known limitation. Just created an issue: https://github.com/AvaloniaUI/Avalonia/issues/18292
